// @contentpi/lib/dist/graphql/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getQueryName = exports.validateFields = exports.getGraphQlError = void 0;
function getGraphQlError(error) {
    return {
        error: true,
        message: error.toString().replace('Error: GraphQL error: ', '')
    };
}
exports.getGraphQlError = getGraphQlError;
function validateFields(validations, values, alert = 'Error trying to create the item') {
    const messages = {};
    Object.keys(validations).forEach(field => {
        const { len, isEmpty } = validations[field];
        if (len && len.arg) {
            if (values[field].length !== len.arg) {
                messages[field] = {
                    msg: len.msg
                };
            }
        }
        if (isEmpty) {
            if (values[field] === '' || !values[field]) {
                messages[field] = {
                    msg: isEmpty.msg
                };
            }
        }
    });
    return Object.keys(messages).length > 0
        ? {
            error: true,
            alert,
            messages
        }
        : {
            error: false
        };
}
exports.validateFields = validateFields;
function getQueryName(query) {
    if (query && query.definitions && query.definitions.length > 0) {
        return query.definitions[0].name.value;
    }
    return false;
}
exports.getQueryName = getQueryName;
