// @contentpi/lib/dist/is/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isLanguage = exports.isEmptyObject = exports.isSymbol = exports.isBrowser = exports.isUndefined = exports.isString = exports.isPasswordMatch = exports.isPassword = exports.isObject = exports.isJson = exports.isFunction = exports.isNumber = exports.isFalse = exports.isDefined = exports.isArray = void 0;
const languages_1 = require("../languages");
function isArray(v) {
    return v instanceof Array;
}
exports.isArray = isArray;
function isDefined(v) {
    return typeof v !== 'undefined' && v !== null;
}
exports.isDefined = isDefined;
function isFalse(v) {
    return isDefined(v) && v === false;
}
exports.isFalse = isFalse;
function isNumber(v) {
    return typeof v === 'number';
}
exports.isNumber = isNumber;
function isFunction(v) {
    return typeof v === 'function';
}
exports.isFunction = isFunction;
function isJson(str) {
    if (!str || str === null) {
        return false;
    }
    try {
        JSON.parse(str);
    }
    catch (e) {
        return false;
    }
    return true;
}
exports.isJson = isJson;
function isObject(v) {
    return isDefined(v) && typeof v === 'object' && !isArray(v);
}
exports.isObject = isObject;
function isPassword(password, min = 8) {
    return Boolean(password && password.length >= min);
}
exports.isPassword = isPassword;
function isPasswordMatch(p1, p2) {
    return isPassword(p1) && isPassword(p2) && p1 === p2;
}
exports.isPasswordMatch = isPasswordMatch;
function isString(v) {
    return isDefined(v) && typeof v === 'string';
}
exports.isString = isString;
function isUndefined(v) {
    return typeof v === 'undefined' || v === null;
}
exports.isUndefined = isUndefined;
function isBrowser() {
    return typeof window !== 'undefined';
}
exports.isBrowser = isBrowser;
function isSymbol(v) {
    return typeof v === 'symbol' || isObject(v);
}
exports.isSymbol = isSymbol;
function isEmptyObject(v) {
    return v ? Object.keys(v).length === 0 : true;
}
exports.isEmptyObject = isEmptyObject;
const isLanguage = (lang) => {
    return !!languages_1.languagesList[lang];
};
exports.isLanguage = isLanguage;
